### PR TITLE
util/slice: migrate test-infra to testify

### LINKED ISSUE #26418: migrate test-infra to testify for `util/slice` pkg

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingy94@gmail.com>

### What problem does this PR solve?

Issue Number: close #26418

### Release note 

- No release note

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
LGTM. Nice work!

**Review by bb7133 — APPROVED** [review]

**Comment by bb7133** [comment]
Thanks for your contribution!

@xhebox PTAL

File: util/slice/slice_test.go [hunk 1]
```diff
@@ -41,10 +34,13 @@ func (s *testSliceSuite) Test(c *C) {
 		{[]int{2, 2, 4}, true, false, true},
 	}
 
-	for _, t := range tests {
-		even := func(i int) bool { return t.a[i]%2 == 0 }
-		c.Assert(AnyOf(t.a, even), Equals, t.anyOf)
-		c.Assert(NoneOf(t.a, even), Equals, t.noneOf)
-		c.Assert(AllOf(t.a, even), Equals, t.allOf)
+	for _, test := range tests {
+		t.Run(strings.Join(strings.Fields(fmt.Sprint(test.a)), ","), func(t *testing.T) {
```
Comment (xhebox) [inline util/slice/slice_test.go:38]: Maybe `fmt.Sprint(test.a)` is enough. Or just `fmt.Sprintf("%d: %s", idx, test.a)`, where idx is `for idx, test := range tests {`.

Comment (kemingy) [reply util/slice/slice_test.go:38]: Thanks for your advice. Will fix this.

**Review by xhebox — APPROVED** [review]

**Pull request merged**